akka-docs/rst/java/code/docs/http/javadsl/server/directives/FutureDirectivesExamplesTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package docs.http.javadsl.server.directives;
+
+import java.util.concurrent.CompletableFuture;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.server.Marshaller;
+import akka.http.javadsl.server.PathMatchers;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.http.scaladsl.model.StatusCodes;
+import akka.japi.pf.PFBuilder;
+import org.junit.Test;
+
+import static scala.compat.java8.JFunction.func;
+
+public class FutureDirectivesExamplesTest extends JUnitRouteTest {
+
+    @Test
+    public void testOnComplete() {
+        //#onComplete
+        // import static scala.compat.java8.JFunction.func;
+
+        final Route route = path(
+                PathMatchers.segment("divide")
+                        .slash(PathMatchers.integerSegment())
+                        .slash(PathMatchers.integerSegment()),
+                (a, b) -> onComplete(
+                        () -> CompletableFuture.supplyAsync(() -> a / b),
+                        maybeResult -> maybeResult
+                                .map(func(result ->
+                                        complete(String.format("The result was %s", result)))
+                                )
+                                .recover(new PFBuilder<Throwable, Route>()
+                                        .matchAny(ex ->
+                                                complete(StatusCodes.InternalServerError(),
+                                                        String.format("An error occurred: %s",
+                                                                ex.getMessage())
```

Comment:
I agree with you. I've updated it to 2 spaces formatting and now is much more readable :)